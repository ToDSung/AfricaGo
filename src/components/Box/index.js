import React, { forwardRef } from 'react';

import PropTypes from 'prop-types';

const Box = forwardRef(function Box({ className, children, ...props }, ref) {
  return (
    <div
      className={`flex items-center justify-center rounded border text-base ${className}`}
      {...props}
      ref={ref}
    >
      {children}
    </div>
  );
});

Box.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

export default Box;
