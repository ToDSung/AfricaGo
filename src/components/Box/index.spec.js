/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';

import { render } from '@testing-library/react';

import Box from '.';

test('should Box render', () => {
  const component = renderer.create(
    <Box>
      <div>box content!</div>
    </Box>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should Box Content render', () => {
  const { queryByText } = render(
    <Box>
      <div>box content!</div>
    </Box>
  );

  expect(queryByText(/box content/i)).toBeTruthy();
});
