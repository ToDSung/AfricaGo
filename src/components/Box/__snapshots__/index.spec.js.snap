// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Box render 1`] = `
<div
  className="flex items-center justify-center rounded border text-base "
>
  <div>
    box content!
  </div>
</div>
`;
